connector/avro/src/main/scala/org/apache/spark/sql/avro/SchemaConverters.scala

```diff
@@ -144,11 +147,28 @@ object SchemaConverters {
           case _ =>
             // Convert complex unions to struct types where field names are member0, member1, etc.
             // This is consistent with the behavior when converting between Avro and Parquet.
+            val use_stable_id = SQLConf.get.getConf(SQLConf.AVRO_STABLE_ID_FOR_UNION_TYPE)
+
+            var fieldNameSet : Set[String] = Set()
             val fields = avroSchema.getTypes.asScala.zipWithIndex.map {
               case (s, i) =>
                 val schemaType = toSqlTypeHelper(s, existingRecordNames)
                 // All fields are nullable because only one of them is set at a time
-                StructField(s"member$i", schemaType.dataType, nullable = true)
+
+                val fieldName = if (use_stable_id) {
+                  // Avro's field name may be case sensitive, so field names for two named type
+                  // could be "a" and "A" and we need to distinguish them.
+                  var temp_name = s"member_${s.getName.toLowerCase(Locale.ROOT)}"
+                  while (fieldNameSet.contains(temp_name)) {
+                    temp_name = s"${temp_name}_$i"
```

Comment:
I was thinking if we could simply throw an error when this case happens; the reason is that they might not be stable identifiers anymore. We could explain that stable identifiers can only be used if the types are unique, if you have more than one type that has the same name, please use sequential numeric ids instead.

Alternatively, we could just store them as is, without converting to upper or lower case - that could be an option.